### ISSUE TITLE
Introspection runs into endless exception reporting

### ISSUE THREAD

**Issue by haraldsteinlechner** [aardvark-platform/aardvark.base#79]
in, pro3d, not all packages are up to date and i missed quite some development there, introspection runs into an endless exception reporting loop. Is this something somebody stumbled across already? it looks like this:
![image](https://user-images.githubusercontent.com/513281/186504401-ca38c1cf-1c47-43aa-ade6-4fe804b69955.png)

**Comment by haraldsteinlechner** [comment]
nope, not goes forever but for several minutes. it seems to happen within custom pickler registration.
![image](https://user-images.githubusercontent.com/513281/186506761-81154ef7-eb08-4ff5-ada5-27f74815e225.png)
https://github.com/pro3d-space/PRo3D/issues/228

**Comment by haraldsteinlechner** [comment]
so in the end, adaptify seems to silently require f# 6.0 at runtime, and our introspection code is not robust to problems like this by printing thousands of errors till it finally crashes.

**Comment by stefanmaierhofer** [comment]
I am implementing bounded, per-query introspection diagnostics and incomplete-scan cache recovery on `fix/introspection-partial-scans`. The change will retain partial successes, avoid writing caches after incomplete scans, retry live when cache decoding fails, and add deterministic regressions plus clean hit/miss performance comparisons. I will open an unmerged PR when validation is complete.

**Comment by stefanmaierhofer** [comment]
Implementation is available in PR #99: https://github.com/aardvark-platform/aardvark.base/pull/99

It retains partial introspection successes, bounds and deduplicates reflection diagnostics, prevents incomplete scans from being cached, and retries incomplete cache decodes live. Local validation passed 1,895 maintained tests with 31 existing skips and no failures; the full build had zero warnings/errors, docs passed, and matched Release benchmarks found no clean-path throughput or allocation regression.
